sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -1126,19 +1126,21 @@ def mean_split(
         ]
         return SplitPrimitiveTensor(ts=shards, shard_dim=shard_dim_new)
     else:
-        gathered = cat(
-            [
-                (
-                    transfer_to_logical_device(shard, x.devices[0])
-                    if i != 0
-                    else barrier_on_logical_device(shard, x.devices[0])
-                )
-                for i, shard in enumerate(x.shards)
-            ],
-            dim=x.shard_dim,
+
+        partial_sums = [
+            sum(shard, dim=dim, keepdim=keepdim, dtype=dtype) for shard in x.shards
+        ]
+        total_sum = sharded_sum(
+            SplitPrimitiveTensor(ts=partial_sums, shard_dim=0, devices=x.devices)
+        )
+
+        total_cnt = math.prod(x.shape[d] for d in dim)
+
+        global_mean = total_sum / total_cnt
+
+        return ReplicatedTensor(
+            ts=global_mean, shard_count=x.shard_count, devices=x.devices
         )
```

Comment:
`sharded_sum` would return a tensor on device 0. Actually we would like to do an `all_reduce`. I understand that due to performance limitation of parallel transfers in IREE this actually does a bit less work as it does the normalization locally on only one device. But ultimately we want an all-reduce. You could at least leave a TODO comment that once IREE can issue parallel transfers on one queue we should do an all-reduce.